src/components/task-plan/homework.cjsx

```diff
@@ -0,0 +1,404 @@
+      return <span className="-no-show"/>
+    unless TocStore.isLoaded()
+      TocActions.load(courseId)
+      return <span className="-loading">Loading...</span>
+
+    chapters = _.map(TocStore.get(), @renderChapterPanels)
+    buttonStyle = if selected?.length then 'primary' else 'disabled'
+    shouldShowExercises = selected?.length and @state.showProblems
+
+    if not @state.showProblems
+      footer =
+        <span>
+          <BS.Button bsStyle={buttonStyle} onClick={@selectProblems}>Show Problems</BS.Button>
+          <BS.Button bsStyle="link" onClick={hide}>Cancel</BS.Button>
+        </span>
+
+    <div>
+      <BS.Panel footer={footer}>
+        <div className="select-reading-modal">
+          {chapters}
+        </div>
+      </BS.Panel>
+      <ExerciseSummary
+        shouldShow={shouldShowExercises} 
+        canReview={true}
+        reviewClicked={hide}
+        planId={planId}/>
+      <AddExercises
+        courseId={courseId} 
+        planId={planId} 
+        shouldShow={shouldShowExercises} 
+        pageIds={selected}
+        hide={@hideSectionTopics}/>
+    </div>
+
+HomeworkPlan = React.createClass
+  getInitialState: ->
+    {showSectionTopics: false}
+
+  setDueAt: (value) ->
+    {id} = @props
+    TaskPlanActions.updateDueAt(id, value)
+
+  setTitle: ->
+    {id} = @props
+    value = @refs.title.getDOMNode().value
+    TaskPlanActions.updateTitle(id, value)
+
+  setDescription: ->
+    {id} = @props
+    desc = @refs.description.getDOMNode().value
+    TaskPlanActions.updateDescription(id, desc)
+  
+  showSectionTopics: ->
+    @setState({
+      showSectionTopics: true
+    })
+
+  hideSectionTopics: ->
+    @setState({
+      showSectionTopics: false
+    })
+
+  render: ->
+    {id, courseId} = @props
+    plan = TaskPlanStore.get(id)
+    description = TaskPlanStore.getDescription(id)
+    headerText = if TaskPlanStore.isNew(id) then 'Add Homework' else 'Edit Homework'
+    topics = TaskPlanStore.getTopics(id)
+    shouldShowExercises = TaskPlanStore.getExercises(id)?.length and not @state?.showSectionTopics
+
+    if plan?.due_at
+      dueAt = new Date(plan.due_at)
+
+    today = new Date(plan.due_at)
+    footer = <PlanFooter id={id} courseId={courseId} clickedSelectProblem={@showSectionTopics}/>
+
+    formClasses = ['create-homework']
+    if @state?.showSectionTopics then formClasses.push('hide')
+
+    if (TaskPlanStore.isPublished(id))
+      dueAtElem = <span>{new moment(dueAt).format('MMM Do, YYYY')}</span>
+    else
+      dueAtElem = <DateTimePicker
+                    id="homework-due-date"
+                    format="MMM dd, yyyy"
+                    time={false}
+                    calendar={true}
+                    readOnly={false}
+                    onChange={@setDueAt}
+                    min={new Date()}
+                    value={dueAt}/>
+
+    <div>
+      <BS.Panel bsStyle='default' className={formClasses.join(' ')} footer={footer}>
+        <h1>{headerText}</h1>
```

Comment:
Why isn't this a prop sent to `BS.Panel` (ie `footer={footer} header={headerText}`) like the reading?